index.js

```diff
@@ -103,8 +104,9 @@ export default class Carousel extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.children !== nextProps.children) {
+    if (!isEqual(this.props.children, nextProps.children)) {
       let childrenLength = 0;
+      this.setState({ currentPage: 1 });
```

Comment:
I think it should be set to 1 either way, because after the children replace the carousel *is* flipped to the first page. It just displays it wrong.

- mount carousel with 3 children
- swipe it to 2nd page
- replace the children with a new set of 3 pages
- the carousel now displays the first page and the info bullet says `2/3`